engine/crates/composition/tests/composition/cost_basic/federated.graphql

```diff
@@ -0,0 +1,45 @@
+directive @core(feature: String!) repeatable on SCHEMA
+
+directive @join__owner(graph: join__Graph!) on OBJECT
+
+directive @join__type(
+    graph: join__Graph!
+    key: String!
+    resolvable: Boolean = true
+) repeatable on OBJECT | INTERFACE
+
+directive @join__field(
+    graph: join__Graph
+    requires: String
+    provides: String
+) on FIELD_DEFINITION
+
+directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+
+directive @join__implements(graph: join__Graph!, interface: String!) repeatable on OBJECT | INTERFACE
+
+directive @join__unionMember(graph: join__Graph!, member: String!) repeatable on UNION
+
+enum join__Graph {
+    A @join__graph(name: "a", url: "http://example.com/a")
+}
+
+scalar Foo @cost(weight: 6)
```

Comment:
Should we add the `@cost` directive declaration to the headers?

Reply:
Good point, added it